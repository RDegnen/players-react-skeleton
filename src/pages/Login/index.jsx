import React from 'react';
import { withRouter } from 'react-router-dom';
import ReactRouterPropTypes from 'react-router-prop-types';
import { FormGroup, Label, Button } from 'reactstrap';

import constants from '../../constants';
import FormContainer from '../../components/FormContainer';

const loginUser = (data, history) =>
  fetch(`${constants.API_URL}/api/login`, {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    body: JSON.stringify(data),
  }).then(res => res.json())
    .then((json) => {
      localStorage.setItem('token', json.token);
      history.push('/roster');
    })
    .catch(err => err);

const formGroups = [
  {
    field: 'email',
    type: 'email',
    id: 'email',
    label: 'Email',
  },
  {
    field: 'password',
    type: 'password',
    id: 'password',
    label: 'Password',
  },
  {
    renderGroup: () => (
      <FormGroup row>
        <Label for="login">Login</Label>
        <Button type="submit" value="Submit" id="login">Submit</Button>
      </FormGroup>
    ),
  },
];

export const Login = ({ history }) => (
  <FormContainer history={history} formGroups={formGroups} submitAction={loginUser} />
);

Login.propTypes = {
  history: ReactRouterPropTypes.history.isRequired,
};

export default withRouter(Login);
