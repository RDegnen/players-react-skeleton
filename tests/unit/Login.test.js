import React from 'react';
import { shallow } from 'enzyme';

import Login from '../../src/pages/Login';

const getComponent = props => shallow(<Login history={{}} {...props} />);

describe('<Login />', () => {
  it('should match snapshot', () => {
    const component = getComponent();
    expect(component).toMatchSnapshot();
  });
});
