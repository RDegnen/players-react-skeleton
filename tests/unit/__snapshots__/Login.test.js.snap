// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> should match snapshot 1`] = `
<FormContainer
  formGroups={
    Array [
      Object {
        "field": "email",
        "id": "email",
        "label": "Email",
        "type": "email",
      },
      Object {
        "field": "password",
        "id": "password",
        "label": "Password",
        "type": "password",
      },
      Object {
        "renderGroup": [Function],
      },
    ]
  }
  history={Object {}}
  submitAction={[Function]}
/>
`;
